Source/Particles/Collision/MCCScattering.H

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2021 Roelof Groenewald
+ *
+ * This file is part of WarpX.
+ *
+ * License: ????
+ */
+#ifndef WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+#define WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+
+#include "Utils/WarpXConst.H"
+
+#include <AMReX_Random.H>
+
+/** \brief Function to perform elastic scattering of a particle in the lab
+ * frame. The particle velocities transformed to the COM frame where a hard
+ * sphere collision occurs. The resulting particle velocities are transformed
+ * back to the lab frame and the input particle's velocity is updated.
```

Comment:
If there's any sort of reference for the overall algorithm, this would be a good place to cite it.

Reply:
Thanks for this comment. I'm planning to add a reference to the documentation explaining the MCC algorithm along with all the parameters involved in it.